update.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+"""
+Dirty website parser to generate Dockerfiles for Matlab's MCR releases.
+
+Each Matlab release name (R2020a etc) gets a branch that contains a single
+Dockerfile and each release version (9.8.0 or 9.8.5 for 9.8 Update 5) becomes
+a tag in that branch. Each variant gets a new commit as well.
+So a sample history for R2020a could look something like this:
+
+ + [R2020a] <9.8.1-core> Auto-Update
+ + <9.8.1> Auto-Update
+ + Merged master
+ | \
+ |  + Update in templates
+ | /
+ + <9.8.0-core> Auto-Update
+ + <9.8.0> Auto-Update
+ + <master> Import
+
+(Circle)CI should then fire a docker build and push for every tag only. Shared
+tags for the major version (e.g. 9.8 always pointing to the latest 9.8 tag are
+done in Circle CI to avoid duplicate builds).
+
+Update.py should be run often enough to catch individual matlab releae updates.
```

Comment:
```suggestion
Update.py should be run often enough to catch individual matlab release updates.
```

I don't get the "semi updating" process yet. Does it mean a potential maintainer regularly runs `update.py` from their machine and then commit and push any changes?

Are tags generated automatically?

and who will run `update.py`?

Reply:
Right now `update.py` is simply generating git tags in separate branches and afterwards outputs necessary git commands to trigger circleCI actually building new images based on git tags.

Example output would be:

```
Adding R2020a/9.8.0
Adding R2020a/9.8.0-core
Adding R2019b/9.7.5
Adding R2019b/9.7.5-core
Adding R2019a/9.6.8
Adding R2019a/9.6.8-core
Adding R2018b/9.5.0
Adding R2018b/9.5.0-core
Adding R2018a/9.4.0
Adding R2018a/9.4.0-core
New tags have been added, verify and update to git with:
git push --all
git push origin 9.4.0-core
git push origin 9.4.0
git push origin 9.5.0-core
git push origin 9.5.0
git push origin 9.6.8-core
git push origin 9.6.8
git push origin 9.7.5-core
git push origin 9.7.5
git push origin 9.8.0-core
git push origin 9.8.0
```

Those last `git push` lines can easily be automated away of course but my experience with Matlab tells me that it will still require minor changes once in a while, so this semi-automated process boils down to:

1. run update.py
2. inspect and potentially fix new releases (minor changes in the URL probably)
3. git push all the things

That shouldn't take more than 5 minutes every other month or so.